source/ed/osm2ed.cpp

```diff
@@ -902,52 +888,62 @@ void PoiHouseNumberVisitor::fill_housenumber(const uint64_t osm_id,
     house_numbers.push_back(OSMHouseNumber(str_to_int(it_hn->second), lon, lat, candidate_way->way_ref));
 }
 
-void PoiHouseNumberVisitor::fill_poi(const u_int64_t osm_id, const CanalTP::Tags& tags,
-        const double lon, const double lat, OsmObjectType osm_relation_type) {
-    if (!parse_pois)
-        return;
-
-    std::string ref_tag = "";
-    for(const auto& tags_type: tags_types){
-        if(tags.find(tags_type) != tags.end()){
-            ref_tag = tags_type;
-            break;
+/**
+ * the first rule that matches OSM object decides POI-type
+ * match : OSM object has all OSM tags required by rule
+ */
+const RuleOsmTag2PoiType* PoiTypeParams::get_applicable_poi_rule(const CanalTP::Tags& tags) const {
+    for (const auto& osm_rule: rules) {
+        auto poi_misses_a_tag = [&](const std::pair<const std::string, std::string>& osm_rule_tag) {
```

Comment:
It may not be the same impact to migrate only boost (if @eturck already has a custom boost version).
But yes, adding a auto is always nicer than a pair :)